### PR TITLE
Feature/jitx 872 code generator

### PR DESCRIPTION
**Pull request by ariel-hi**
JITX-872
Generic Code Generator for writing JITX files. This is useful for debugging generated microcontrollers.

Changes:
- New `CodeGenerator` structure with numerous methods to add JITX statements. Builds a `StringBuffer` that holds the file contents.
- Added code generation calls to `micro-controllers`
- New `stm-to-jitx-code` that calls the code generator on the contents of an STM struct. Reproduces a fair bit of the work done in `stm-to-jitx`.
- Minor edits to `stm-to-jitx` to clean up and code change private functions to `protected`.

TODO (in an upcoming PR) :
- Testing beyond calls to `micro-controller`

### CODE REVIEW

**Comment by ariel-hi** [comment]
Here's a sample file generated for part `STM32L031E4Y6`:
[STM32L031E4Y6.stanza.zip](https://github.com/JITx-Inc/open-components-database/files/7282881/STM32L031E4Y6.stanza.zip)

**Comment by ariel-hi** [comment]
Questions I have:
- (Big) Does this belong in JITX or OCDB? Currently its only purpose is to work alongside OCDB files, but I can see this expanding to other use cases such as CAD tooling.
- (Small) Can you see yourself loading the generated JITX code or just for reference on defined supports statements? Adjacently, should the generated components and modules be `public`?

**Comment by ariel-hi** [comment]
Finally, this branch requires changes on JITX that aren't yet integrated, so I expect CI to fail.

File: utils/code-generator.stanza [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+  
+  ;Begin with a new line.
+  defn lnwrite0 (s) : lnprintln(buf0, s)
+  defn lnwrite2 (s) : lnprintln(buf2, s)
+  defn lnwrite4 (s) : lnprintln(buf4, s)
+  defn lnwrite6 (s) : lnprintln(buf6, s)
+
+  new CodeGenerator :
+    defmethod write-added-syntaxes (this, syntaxes:Tuple<Syntax>) :
+      write0("#use-added-syntax(%,)" % [seq(to-string, syntaxes)])
```
Comment (PhilippeFerreiraDeSousa) [inline utils/code-generator.stanza:83]: `%,` does `to-string` already:
```
write0("#use-added-syntax(%,)" % [syntaxes])
```

**Pull request closed without merging**